qiskit/providers/aer/jobs/utils.py

```diff
@@ -55,35 +56,139 @@ def wrapper(*args, **kw):
-            None don't split (Default: None).
-        qobj_id (str): Optional, set a fixed qobj ID for all subjob qobjs.
+    num_shot_jobs, shot_mod = divmod(qobj.config.shots, max_shot_size)
+    qobj_list = []
 
-    Returns:
-        List: A list of qobjs.
-    """
+    if shot_mod == 0 and num_shot_jobs == 1:
+        return qobj
+
+    if shot_mod > 0:
+        setattr(qobj.config, "shots", shot_mod)
+        for experiment in qobj.experiments:
+            _id = str(uuid.uuid4())
+            setattr(experiment.header, "metadata", {"id": _id})
+        qobj_list.append(qobj)
+
+    if num_shot_jobs > 1:
+        _qid = qobj_id or str(uuid.uuid4())
+        _config = copy.deepcopy(qobj.config)
+        setattr(_config, "shots", max_shot_size)
+        experiment_list = []
+        for experiment in qobj.experiments:
+            _id = str(uuid.uuid4())
+            for _ in range(num_shot_jobs):
+                cpy_exp = copy.deepcopy(experiment)
+                setattr(cpy_exp.header, "metadata", {"id": _id})
+                experiment_list.append(cpy_exp)
+        qobj_list.append(QasmQobj(_qid, _config, experiment_list, qobj.header))
+
+    return qobj_list
+
+
+def _split_qobj(qobj, max_size, qobj_id, seed):
     # Check if we don't need to split
     if max_size is None or not max_size > 0:
-        return qobj
+        return qobj, seed
+
     num_jobs = ceil(len(qobj.experiments) / max_size)
     if num_jobs == 1:
-        return qobj
+        return qobj, seed
 
+    qobjs = []
     # Check for parameterizations
     params = getattr(qobj.config, 'parameterizations', None)
-    qobjs = []
+
+    exp_id = None
+    shift_index = 0
+    seed_shift = 256
+    _seed = 0
     for i in range(num_jobs):
         sub_id = qobj_id or str(uuid.uuid4())
         indices = slice(i * max_size, (i + 1) * max_size)
         sub_exp = qobj.experiments[indices]
         sub_config = qobj.config
+
         if params is not None:
-            sub_config = copy.copy(qobj.config)
             sub_config.parameterizations = params[indices]
+            sub_config = copy.copy(qobj.config)
+
+        if seed > 0:
+            id_dat = getattr(sub_exp[0].header, "metadata", None)
+            if id_dat is not None and "id" in id_dat:
+                _id = id_dat["id"]
+                if _id == exp_id:
+                    shift_index = shift_index + 1
+                else:
+                    exp_id = _id
+                    shift_index = 0
+                _seed = seed + seed_shift * shift_index
+            if sub_config is qobj.config:
+                sub_config = copy.copy(qobj.config)
+            setattr(sub_config, "seed_simulator", _seed)
+
         qobjs.append(type(qobj)(sub_id, sub_config, sub_exp, qobj.header))
-    return qobjs
+
+    if seed > 0:
+        seed = _seed + seed_shift
+
+    return qobjs, seed
+
+
+def _check_custom_instruction(experiments, custom_instructions):
```

Comment:
This is a slow check since it has to iterate over all circuits. We should really build this instruction map as part of `assemble` which already iterates over all instructions in a circuit.

Reply:
Agree, but I believe that no function to return an instruction map in `assemble`. So, the AerBackend needs to iterate instructions at least once.